lms/djangoapps/show_answer/show_answer_field_override.py

```diff
@@ -0,0 +1,39 @@
+"""
+FieldOverride that forces
+"""
+
+
+from django.conf import settings
+
+from common.lib.xmodule.xmodule.capa_base import SHOWANSWER
+from lms.djangoapps.courseware.field_overrides import FieldOverrideProvider
+from openedx.features.course_experience import RELATIVE_DATES_FLAG
+
+class ShowAnswerFieldOverride(FieldOverrideProvider):
+    """
+    A concrete implementation of
+    :class:`~courseware.field_overrides.FieldOverrideProvider` which forces
+
+    """
+    def get(self, block, name, default):
+        if name != 'showanswer' or not self.fallback_field_data.has(block, 'showanswer'):
+            return default
+
+        mapping = {
+            SHOWANSWER.ATTEMPTED: SHOWANSWER.ATTEMPTED_NO_PAST_DUE,
+            SHOWANSWER.CLOSED: SHOWANSWER.AFTER_ALL_ATTEMPTS,
+            SHOWANSWER.CORRECT_OR_PAST_DUE: SHOWANSWER.ANSWERED,
+            SHOWANSWER.FINISHED: SHOWANSWER.AFTER_ALL_ATTEMPTS_OR_CORRECT,
+            SHOWANSWER.PAST_DUE: SHOWANSWER.NEVER,
+        }
+        current_show_answer_value = self.fallback_field_data.get(block, 'showanswer')
+
+        if current_show_answer_value in mapping:
+            return mapping[current_show_answer_value]
+
+        return default
```

Comment:
Could make this just `return mapping.get(current_show_answer_value, default)`.

That said, I think the default value for `showanswer` is actually `finished`, which we would still need to map, so you might need to do something like: `return mapping.get(current_show_answer_value, mapping.get(default, default))`

Reply:
So the `default` is actually weird for this function. It doesn't represent the default value for the field, but instead is a default value passed into the function. It is called `NOTSET` and is just an empty object (`object()`). And from testing, although some blocks do not have the showanswer attribute, they are still being correctly mapped (I _think_ because of an earlier pass at the course level and then inheritance).

Reply:
Yeah, I didn't remember when inheritance was happening. I *think* that the explanation might be that the fallback field data provider is doing the inheritance, and then this code is mapping the inherited value through the mapper. Which maybe means we're mapping more that we need to, but the mapping should be inexpensive.